src/thread_pool.rs

```diff
@@ -164,16 +199,30 @@ impl Registry {
                 return Work::None;
             }
 
+            self.worker_idle.notify_all();
```

Comment:
It's probably worth a comment in the `state.injected_jobs.pop_front()` section on line 192 above indicating why we choose not to notify `worker_idle` -- in particular, although popping from `injected_jobs` may cause the `worker_idle` condition to become true, incrementing `threads_at_work` will ensure it is false.